File: scripts/standardize_titles.js
const fs = require('fs');
const path = require('path');

const rootDir = path.dirname(__dirname);
const skipDirs = new Set(['.git', 'assets', 'scripts']);
const skipFiles = new Set([
    '404.html',
    'privacy-policy/index.html',
    'terms-of-service/index.html',
    'dmca/index.html',
    'about-us/index.html',
    'contact-us/index.html'
]);

let updatedCount = 0;
let skippedCount = 0;

function walkDir(currentPath) {
    const files = fs.readdirSync(currentPath);
    
    for (const file of files) {
        const filePath = path.join(currentPath, file);
        const stats = fs.statSync(filePath);
        const relPath = path.relative(rootDir, filePath).replace(/\\/g, '/');
        
        if (stats.isDirectory()) {
            if (skipDirs.has(file)) continue;
            walkDir(filePath);
        } else if (file.endsWith('.html')) {
            if (skipFiles.has(relPath)) {
                // console.log(`Skipping system/legal page: ${relPath}`);
                skippedCount++;
                continue;
            }
            processFile(filePath, relPath);
        }
    }
}

function processFile(filePath, relPath) {
    try {
        let content = fs.readFileSync(filePath, 'utf8');
        
        // 1. Try to find <h1 class="game-title"> (most specific)
        let gameName = null;
        const h1GameTitle = content.match(/<h1[^>]*class=["']game-title["'][^>]*>([\s\S]*?)<\/h1>/i);
        if (h1GameTitle) {
            gameName = h1GameTitle[1];
        } else {
            // 2. Try to find <h1> (fallback for index/category pages)
            const h1Any = content.match(/<h1>([\s\S]*?)<\/h1>/i);
            if (h1Any) gameName = h1Any[1];
        }
        
        if (gameName) {
            // Clean gameName: remove HTML tags, trim, and collapse whitespace
            gameName = gameName.replace(/<[^>]*>/g, '').trim().replace(/\s+/g, ' ');
            
            // Escape double quotes for meta tag content
            const metaGameName = gameName.replace(/"/g, '&quot;');
            
            let newContent = content;
            
            // Update <title>
            newContent = newContent.replace(/<title>[\s\S]*?<\/title>/i, `<title>${gameName}</title>`);
            
            // Update og:title
            newContent = newContent.replace(/<meta property=["']og:title["'] content=["'].*?["']>/gi, `<meta property="og:title" content="${metaGameName}">`);
            
            // Update twitter:title
            newContent = newContent.replace(/<meta name=["']twitter:title["'] content=["'].*?["']>/gi, `<meta name="twitter:title" content="${metaGameName}">`);
            
            if (newContent !== content) {
                fs.writeFileSync(filePath, newContent, 'utf8');
                console.log(`Updated: ${relPath} -> Title: '${gameName}'`);
                updatedCount++;
            }
        } else {
            // console.warn(`Warning: No <h1> or game-title found in ${relPath}`);
        }
    } catch (err) {
        console.error(`Error processing ${relPath}: ${err.message}`);
    }
}

console.log('Starting title standardization (Node.js version)...');
walkDir(rootDir);
console.log(`\nStandardization complete.`);
console.log(`Updated: ${updatedCount} files`);
console.log(`Skipped: ${skippedCount} system/legal files`);

File: scripts/standardize_titles.py
import os
import re

def standardize_titles():
    # Root directory is one level up from the scripts directory
    root_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    
    # Files/Dirs to skip
    skip_dirs = {'.git', 'assets', 'scripts'}
    skip_files = {
        '404.html', 
        'privacy-policy/index.html', 
        'terms-of-service/index.html', 
        'dmca/index.html', 
        'about-us/index.html', 
        'contact-us/index.html'
    }
    
    updated_count = 0
    skipped_count = 0
    
    for root, dirs, files in os.walk(root_dir):
        # Skip specified directories
        dirs[:] = [d for d in dirs if d not in skip_dirs]
        
        for file in files:
            if not file.endswith('.html'):
                continue
                
            file_path = os.path.join(root, file)
            # Use forward slashes for cross-platform matching
            rel_path = os.path.relpath(file_path, root_dir).replace('\\', '/')
            
            if rel_path in skip_files:
                print(f"Skipping system/legal page: {rel_path}")
                skipped_count += 1
                continue
                
            try:
                with open(file_path, 'r', encoding='utf-8') as f:
                    content = f.read()
            except Exception as e:
                print(f"Error reading {rel_path}: {e}")
                continue
                
            # 1. Try to find <h1 class="game-title"> (most specific)
            game_name = None
            match = re.search(r'<h1[^>]*class=["\']game-title["\'][^>]*>(.*?)</h1>', content, re.DOTALL | re.IGNORECASE)
            if match:
                game_name = match.group(1)
            else:
                # 2. Try to find <h1> (for category/index pages)
                match = re.search(r'<h1>(.*?)</h1>', content, re.DOTALL | re.IGNORECASE)
                if match:
                    game_name = match.group(1)
            
            if game_name:
                # Strip HTML tags from game_name (e.g., <i>, <span>, etc.)
                game_name = re.sub(r'<[^>]*>', '', game_name).strip()
                # Replace multiple whitespace/newlines with a single space
                game_name = re.sub(r'\s+', ' ', game_name)
                
                # Escape double quotes for meta tag content
                meta_game_name = game_name.replace('"', '&quot;')
                
                new_content = content
                
                # Update <title>
                new_content = re.sub(r'<title>.*?</title>', f'<title>{game_name}</title>', new_content, flags=re.DOTALL | re.IGNORECASE)
                
                # Update og:title
                new_content = re.sub(r'<meta property=["\']og:title["\'] content=["\'].*?["\']>', f'<meta property="og:title" content="{meta_game_name}">', new_content, flags=re.IGNORECASE)
                
                # Update twitter:title
                new_content = re.sub(r'<meta name=["\']twitter:title["\'] content=["\'].*?["\']>', f'<meta name="twitter:title" content="{meta_game_name}">', new_content, flags=re.IGNORECASE)

                if new_content != content:
                    try:
                        with open(file_path, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        print(f"Updated: {rel_path} -> Title: '{game_name}'")
                        updated_count += 1
                    except Exception as e:
                        print(f"Error writing {rel_path}: {e}")
                else:
                    # print(f"No changes needed: {rel_path}")
                    pass
            else:
                print(f"Warning: No <h1> or game-title found in {rel_path}")

    print(f"\nStandardization complete.")
    print(f"Updated: {updated_count} files")
    print(f"Skipped: {skipped_count} files")

if __name__ == "__main__":
    standardize_titles()
